pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -737,23 +762,15 @@ func (a *FlavorAssigner) fitsResourceQuota(log logr.Logger, fr resources.FlavorR
 		return fit, borrow, nil
 	}
 
-	// Check if preemption is possible
-	mode := noFit
-	// For single-level hierarchies, mayReclaimInHierarchy = true iff val <= rQuota.Nominal
-	preemptionPossibility := a.oracle.SimulatePreemption(log, a.cq, *a.wl, fr, val)
-	if val <= rQuota.Nominal || mayReclaimInHierarchy {
-		mode = preempt
-		if preemptionPossibility == preemptioncommon.Reclaim {
-			mode = reclaim
-		}
-	} else if a.canPreemptWhileBorrowing() && (preemptionPossibility != preemptioncommon.NoCandidates) {
```

Comment:
The PR does not actually add a unit test for the logic change when `val <= rQuota.Nominal || mayReclaimInHierarchy` is false, but `a.canPreemptWhileBorrowing()` is true. 

I add it in the follow up: https://github.com/kubernetes-sigs/kueue/pull/5735/files#diff-130e36b28b6a84d355879caa6b7927c98ec18c63adad6ab4fc486458763080bfR2797